src/Halogen/Query/HalogenM.purs

```diff
@@ -163,8 +163,16 @@ getRef p = HalogenM $ liftF $ GetRef p identity
 raise :: forall s act ps o m. o -> HalogenM' s act ps o m Unit
 raise o = HalogenM $ liftF $ Raise o unit
 
-fork :: forall s act ps o m a. MonadAff m => HalogenM' s act ps o m a -> HalogenM' s act ps o m (Error -> m Unit)
-fork a = map liftAff <$> HalogenM (liftF $ Fork $ FF.fork a)
+newtype ForkId = ForkId Int
+
+derive newtype instance eqForkId :: Eq ForkId
+derive newtype instance ordForkId :: Ord ForkId
+
+fork :: forall s act ps o m. MonadAff m => HalogenM' s act ps o m Unit -> HalogenM' s act ps o m ForkId
```

Comment:
Would be nice if you add a comment to this function #546

Reply:
Can't you remove the `MonadAff` constraint now?